pkg/k8s/resource/resource.go

```diff
@@ -536,21 +570,28 @@ func (r *resource[T]) Events(ctx context.Context, opts ...EventsOpt) <-chan Even
 // e.g. it implements the eventDone() method called by Event[T].Done().
 type keyQueue struct {
 	workqueue.RateLimitingInterface
-	errorHandler ErrorHandler
+	errorHandler   ErrorHandler
+	deletedObjects sync.Map
 }
 
 func (kq *keyQueue) AddSync() {
 	kq.Add(syncEntry{})
 }
 
 func (kq *keyQueue) AddUpsert(key Key) {
+	kq.deletedObjects.Delete(key)
+
 	// The entries must be added by value and not by pointer in order for
 	// them to be compared by value and not by pointer.
-	kq.Add(upsertEntry{key})
+	kq.Add(upsertOrDeleteEntry{key})
```

Comment:
I would suggest to leave another comment in here explaining why an "Add" is passing an "upsertOrDelete" entry

Reply:
`kq` is the queue, so we're adding into the queue. It's not `Add` of an object, but it's an add of a workqueue entry.

Reply:
> Or what do you think about switching to Enqueue instead? It won't match the workqueue API it's wrapping, but it would be less confusing.

Oh, I think this would definitely sound less confusing :)